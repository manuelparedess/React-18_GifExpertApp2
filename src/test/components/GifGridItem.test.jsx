const { render, screen } = require("@testing-library/react")
const { GifGridItem } = require("../../components/GifGridItem")


describe('Pruebas en GifGridItem', () => { 
    
    const title = "Real Madrid";
    const url = "https://realmadrid.com/real-madrid.jpg";

    test('Debe mostrarse correctamente', () => {  

        const { container } = render( <GifGridItem title={ title } url={ url } /> );

        expect(container).toMatchSnapshot();
    })

    test('debe tener la imagen igual al url y alt de los props', () => { 

        render( <GifGridItem title={ title } url={ url } /> );
        //screen.debug();

        expect( screen.getByRole('img').src ).toBe(url);
        expect( screen.getByRole('img').alt ).toBe(title);

    })

    test('debe tener un parrafo con el titulo', () => { 

        render( <GifGridItem title={ title } url={ url } /> );

        expect( screen.getByText(title) ).toBeTruthy();

    })

})