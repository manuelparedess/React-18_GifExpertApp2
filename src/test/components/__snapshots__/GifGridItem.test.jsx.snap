// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe mostrarse correctamente 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Real Madrid"
      src="https://realmadrid.com/real-madrid.jpg"
    />
    <p>
      Real Madrid
    </p>
  </div>
</div>
`;
